llvm/lib/Target/RISCV/RISCVInstrInfoP.td

```diff
@@ -0,0 +1,1050 @@
+def PAADD_W  : RVPBinary1LongFW<0b0011, 0b01, 0b000, "paadd.w">;
+
+def PSADDU_W : RVPBinary1LongFW<0b0110, 0b01, 0b000, "psaddu.w">;
+def PAADDU_W : RVPBinary1LongFW<0b0111, 0b01, 0b000, "paaddu.w">;
+
+def PSUB_W   : RVPBinary1LongFW<0b1000, 0b01, 0b000, "psub.w">;
+def PSSUB_W  : RVPBinary1LongFW<0b1010, 0b01, 0b000, "pssub.w">;
+def PASUB_W  : RVPBinary1LongFW<0b1011, 0b01, 0b000, "pasub.w">;
+
+def PSSUBU_W : RVPBinary1LongFW<0b1110, 0b01, 0b000, "pssubu.w">;
+def PASUBU_W : RVPBinary1LongFW<0b1111, 0b01, 0b000, "pasubu.w">;
+} // Predicates = [HasStdExtP, IsRV64]
+
+
+let Predicates = [HasStdExtP] in {
+def SLX         : RVPBinary1LongFW<0b0001, 0b11, 0b001, "slx">;
+def PMUL_H_B01  : RVPBinary1LongFW<0b0010, 0b00, 0b001, "pmul.h.b01">;
+
+def MVM         : RVPBinary1LongFW<0b0101, 0b00, 0b001, "mvm">;
+def MVMN        : RVPBinary1LongFW<0b0101, 0b01, 0b001, "mvmn">;
+def MERGE       : RVPBinary1LongFW<0b0101, 0b10, 0b001, "merge">;
+def SRX         : RVPBinary1LongFW<0b0101, 0b11, 0b001, "srx">;
+def PMULU_H_B01 : RVPBinary1LongFW<0b0110, 0b00, 0b001, "pmulu.h.b01">;
+def PDIFSUMU_B  : RVPBinary1LongFW<0b0110, 0b10, 0b001, "pdifsumu.b">;
+def PDIFSUMAU_B : RVPBinary1LongFW<0b0111, 0b10, 0b001, "pdifsumau.b">;
+} // Predicates = [HasStdExtP]
+
+let DecoderNamespace = "RISCV32Only_",
+    Predicates = [HasStdExtP, IsRV32] in {
+def MUL_H01   : RVPBinary1LongFW<0b0010, 0b01, 0b001, "mul.h01">;
+def MACC_H01  : RVPBinary1LongFW<0b0011, 0b01, 0b001, "macc.h01">;
+
+def MULU_H01  : RVPBinary1LongFW<0b0110, 0b01, 0b001, "mulu.h01">;
+def MACCU_H01 : RVPBinary1LongFW<0b0111, 0b01, 0b001, "maccu.h01">;
+} // Predicates = [HasStdExtP, IsRV32]
+
+let Predicates = [HasStdExtP, IsRV64] in {
+def PMUL_W_H01   : RVPBinary1LongFW<0b0010, 0b01, 0b001, "pmul.w.h01">;
+def MUL_W01      : RVPBinary1LongFW<0b0010, 0b11, 0b001, "mul.w01">;
+def PMACC_W_H01  : RVPBinary1LongFW<0b0011, 0b01, 0b001, "pmacc.w.h01">;
+def MACC_W01     : RVPBinary1LongFW<0b0011, 0b11, 0b001, "macc.w01">;
+
+def PMULU_W_H01  : RVPBinary1LongFW<0b0110, 0b01, 0b001, "pmulu.w.h01">;
+def MULU_W01     : RVPBinary1LongFW<0b0110, 0b11, 0b001, "mulu.w01">;
+def PMACCU_W_H01 : RVPBinary1LongFW<0b0111, 0b01, 0b001, "pmaccu.w.h01">;
+def MACCU_W01    : RVPBinary1LongFW<0b0111, 0b11, 0b001, "maccu.w01">;
+} // Predicates = [HasStdExtP, IsRV64]
+
+
+let Predicates = [HasStdExtP] in {
+def PSH1ADD_H   : RVPBinary1F0W<0b010, 0b00, "psh1add.h", 0b010>;
+def PSSH1SADD_H : RVPBinary1F0W<0b011, 0b00, "pssh1sadd.h", 0b010>;
+} // Predicates = [HasStdExtP]
+
+let DecoderNamespace = "RISCV32Only_",
+    Predicates = [HasStdExtP, IsRV32] in {
+def SSH1SADD : RVPBinary1F0W<0b010, 0b01, "ssh1sadd", 0b010>;
+} // Predicates = [HasStdExtP, IsRV32]
+let Predicates = [HasStdExtP, IsRV64] in {
+def PSH1ADD_W   : RVPBinary1F0W<0b010, 0b01, "psh1add.w", 0b010>;
+def PSSH1SADD_W : RVPBinary1F0W<0b011, 0b01, "pssh1sadd.w", 0b010>;
+
+def UNZIP8P   : RVPBinary1F0W<0b110, 0b00, "unzip8p", 0b010>;
```

Comment:
Can we put the opcode string at the end to line up the columns between instructions